### PR TITLE
Blaze: Add payment info screen

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #11702 

## Description

This PR adds a new screen for payment info after confirming details for a Blaze campaign. The view receives campaign details from the creation form and lets the user select a payment method before proceeding to create a campaign.

Changes:
- Added `CreditCardType` for displaying card types on the UI. We are reusing the same images as WCPay for simplicity - note that the icon for the Visa card is not the same as on the design. I hope it's ok.
- Added navigation from the campaign creation form to the payment info screen.
- Added view and view model for the payment info screen. This includes only fetching and displaying card details for now. Due to the size of this PR, I'll add error handling and unit tests in a separate PR.

## Testing instructions

- Log in to a store that's eligible for Blaze.
- If the campaign creation entry point is available on the Dashboard screen, select it and select a product if needed. Otherwise, navigate to the Products tab, select a published product and tap Promote with Blaze.
- On the campaign creation form, tap Confirm Details.
- Confirm that the payment info screen works correctly. 
- Comment out the code to keep `isFetchingPaymentInfo` as `true` and `selectedPaymentMethod` as nil. The activity indicator should be displayed for the loading state.

## Screenshots

Loading state | Result state
--- | --- 
 | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/181">17.0</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/Model/CreditCardType.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            return iconName(suffix: "interac")
+        case .jcb:
+            return iconName(suffix: "jcb")
+        case .diners:
+            return iconName(suffix: "diners")
+        case .unionpay:
+            return iconName(suffix: "unionpay")
+        default:
+            return iconName(suffix: "unknown")
+        }
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Model/CreditCardType.swift]: Suggestion: Can we use `rawValue` instead? 
```suggestion
        iconName(suffix: rawValue)
```

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+            comment: "Button for adding a payment method on the Payment screen in the Blaze campaign creation flow"
+        )
+        static let loading = NSLocalizedString(
+            "blazeConfirmPaymentView.loading",
+            value: "Loading payment methods...",
+            comment: "Text for the loading state on the Payment screen in the Blaze campaign creation flow"
+        )
+        static let agreement = NSLocalizedString(
+            "blazeConfirmPaymentView.agreement",
+            value: "By clicking \"Submit campaign\" you agree to the %1$@ and " +
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift]: Nit: Use "Campaign" to match the CTA button title. 
```suggestion
            value: "By clicking \"Submit Campaign\" you agree to the %1$@ and " +
```

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    var cardDetailView: some View {
+        Button {
+            // TODO: show payment method list
+        } label: {
+            if let icon = viewModel.cardIcon {
+                Image(uiImage: icon)
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 35 * scale)
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift]: nit: Should we move the width value to Layout enum?

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+            await viewModel.updatePaymentInfo()
+        }
+    }
+}
+
+private extension BlazeConfirmPaymentView {
+    var totalAmountView: some View {
+        VStack(alignment: .leading, spacing: Layout.contentPadding) {
+            Text(Localization.paymentTotals)
+                .bold()
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift]: Should this be semibold? 
image

**Review by selanthiraiyan — APPROVED** [review]
Looks great. 🚀 

I left a few nonblocking questions and suggestions.

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift (hunk 4 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift]: Nice catch! Updated in efcd530a05064b1c588d4c7d5192e6de773c66bf.

File: WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift (hunk 3 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentView.swift]: Done in efcd530a05064b1c588d4c7d5192e6de773c66bf.

**Pull request merged**